programs/voter-stake-registry/src/instructions/stake.rs

```diff
@@ -53,10 +53,12 @@ pub fn stake(
         .amount_deposited_native
         .checked_add(amount)
         .ok_or(VsrError::ArithmeticOverflow)?;
+    target.delegate_mining = ctx.accounts.delegate_mining.key();
```

Comment:
as for me better to check then to assign here

Reply:
You can pass an arbitrary account, can't you? What will that check be checking?